src/main/java/core/basesyntax/FieldXY.java

```diff
@@ -13,7 +13,68 @@
 public class FieldXY {
 
     public void moveRobot(Robot robot, int toX, int toY) {
-        //your code
+        Direction neededDirection = robot.getX() < toX ? Direction.RIGHT : Direction.LEFT;
+        if (!robot.getDirection().equals(neededDirection)) {
+            setNeededDirection(robot, neededDirection);
+        }
+        while (robot.getX() != toX) {
+            robot.stepForward();
+        }
+        neededDirection = robot.getY() < toY ? Direction.UP : Direction.DOWN;
+        if (!robot.getDirection().equals(neededDirection)) {
+            setNeededDirection(robot, neededDirection);
+        }
+        while (robot.getY() != toY) {
+            robot.stepForward();
+        }
+    }
+
+    private void setNeededDirection(Robot robot, Direction neededDirection) {
```

Comment:
try to optimize this method it's too long

Reply:
done